extensions/amp-lightbox/1.0/base-element.js

```diff
@@ -33,24 +34,48 @@ export class BaseElement extends PreactBaseElement {
   /** @override */
   init() {
     return dict({
-      'onBeforeOpen': this.toggle_.bind(this, true),
-      'onAfterClose': this.toggle_.bind(this, false),
+      'onBeforeOpen': () => this.beforeOpen_(),
+      'onAfterOpen': () => this.afterOpen_(),
```

Comment:
Do we only need `onAfterOpen` because the element is not rendered when `!mounted`? Can we do `<ContainWrapper style={mounted && {display: 'none'}}>` instead of `mounted && <ContainWrapper>`?

Reply:
We could do `display: none`, but the goal was specifically to not render contents when the lightbox is not open to make it more React-style.

Reply:
To add to this. I generally like the idea that the React component looks more inline with that style. And the cost of an event handler seems not too much of a price to pay?

Reply:
I'm also confused why we can't `setAsContainer` in `onBeforeOpen`. Shouldn't it handle when the scroller eventually becomes visible?

Maybe we shouldn't be calling `onBeforeOpen` until it's actually mounted?

Reply:
It's due to this call:

```
const scroller = this.element.shadowRoot.querySelector('[part=scroller]');
this.setAsContainer(scroller);
```

The lightbox itself is not rendered until it's open and thus `<div part="scroller">` is not loaded either. So doing that `querySelector` will return `null`. For the container it's important to use the actual element with `overflow:auto|scroll` for the intersection observer to provide the accurate values.

Reply:
I will merge before the long weekend. But we will have another opportunity to discuss this with the sidebar component.

Reply:
Sidebar recently replaced the `mounted && <ContainWrapper>` with the stylized mounting approach in #33244.

Reply:
Yep. Found it as well. And I think that's ok - these components are not exactly the same. Though I'd also like to understand the new sidebar a bit better.